Source/CombatExtended/CombatExtended/Comps/CompCIWSTarget.cs

```diff
@@ -0,0 +1,84 @@
+﻿using RimWorld;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using UnityEngine;
+using Verse;
+
+namespace CombatExtended
+{
+    /// <summary>
+    /// Base class for third-party mods compatibility (if their skyfallers\projectiles needs custom targeting logic)
+    /// </summary>
+    public abstract class CompCIWSTarget : ThingComp
+    {
+        protected static Dictionary<Map, List<Thing>> CIWSTargets = new Dictionary<Map, List<Thing>>();
+        public static IEnumerable<Thing> Targets(Map map)
+        {
+            CIWSTargets.TryGetValue(map, out var targets);
+            return targets ?? Enumerable.Empty<Thing>();
+        }
+        public static IEnumerable<Thing> Targets<T>(Map map) where T : CompCIWSTarget
+        {
+            return Targets(map).Where(x => x.HasComp<T>());
+        }
+
+        public CompProperties_CIWSTarget Props => props as CompProperties_CIWSTarget;
```

Comment:
suggestion: props getters should always use a `(cast)` expression since a type mismatch here would be an unrecoverable error. that also reduces the chance of people then treating the props as nullable and adding unnecessary null checks elsewhere.

Reply:
done